doc/elements-confidential-transactions.md

```diff
@@ -0,0 +1,480 @@
+input assets to the output asset must be an “onto” function, or a
+“surjection”). Every blinded transaction output has a Surjection
+proof.
+
+### Elliptic Curve Diffie-Hellman (ECDH)
+
+The receiver and the sender need a shared secret to blind and unblind
+the outputs and to create the range proof. ECDH key exchange is used
+to create the shared secret. When generating a Confidential Address,
+the receiver creates a private blinding key and embeds the public key
+in the address. The sender generates an ephemeral keypair, and with
+the address’ public blinding key, uses ECDH to generate a shared
+secret. The sender’s key for the ECDH key exchange is written into the
+“ConfidentialNonce” field of the transaction output.
+
+See the “Confidential Assets” paper [3] for more detailed information
+and references.
+
+## Transaction Layout and Size Considerations
+
+The layout and fields of a Confidential Transaction are described in
+[elements-tx-format.md](./elements-tx-format.md).
+
+Confidential Transactions are significantly larger than
+non-confidential transactions due to the addition of the range and
+surjection proofs.  Each blinded output adds about 4k bytes to the
+transaction. As the proofs are stored in output witness fields, each
+blinded output adds about 1k vBytes to the virtual size of the
+transaction (please see https://en.bitcoin.it/wiki/Weight_units for
+information about transaction size measurements).
+
+### Range Proofs
+
+A range proof proves that a value lies within a specific range. The
+exact range is determined by the parameters of the range proof. A
+tradeoff exists between the size of the range, and the size of the
+proof’s serialization in the transaction output.  A larger range
+requires a larger proof. In Elements, the default size of the range
+proof is 52 bits, allowing a value up to 2^52-1 satoshis to be
+represented. In Elements, it is possible to create more than 21
+million satoshis of an asset over multiple issuances, but it is not
+possible to send more than 21 million in any one output.  The range
+proof parameters are not part of consensus, and may be overridden and
+adjusted using the `ct_bits` elements configuration parameter. Reducing
+the number of bits will reduce the size of a transaction, and  also
+reduce the maximum provable value of any output. 
+
+An Elements range proof is a Borromean ring signature [4] over
+possible values of each digit in the base 4 representation of an
+output value. Each digit requires the storage of 4+1 elliptic curve
+points. Not including a fixed size header for the range proof, the
+space required for a range proof in Elements is approximately 80 bytes
+per bit of precision (default 52). 
+
+The [secp256k1-zkp range proof implementation](https://github.com/BlockstreamResearch/secp256k1-zkp/blob/master/include/secp256k1_rangeproof.h)
+supports a maximum range of 64 bits, which requires a maximum of 5134 bytes. 
```

Comment:
What is the 5134 bytes here referring to? Are these the vBytes in the transaction? Whereas "64 bits" is the size of the number over which the proof is being done?

Reply:
They are witness bytes, not vbytes, and yes, 64 bits is the size of the number.

Reply:
I'm listing the actual sizes of the objects in bytes. The first paragraph says that it's part of witness fields.